plugins/processors/t128_transform/t128_transform.go

```diff
@@ -105,11 +120,31 @@ func (r *T128Rate) Apply(in ...telegraf.Metric) []telegraf.Metric {
 	return in
 }
 
-func (r *T128Rate) Init() error {
+func (r *T128Transform) Init() error {
 	if len(r.Fields) == 0 {
 		return fmt.Errorf("at least one value must be specified in the 'fields' list")
 	}
 
+	switch r.Transform {
+	case "diff":
+		r.transform = func(t1, t2 time.Time, v1, v2 float64) (float64, error) {
+			return v2 - v1, nil
+		}
+	case "rate":
+		r.transform = func(t1, t2 time.Time, v1, v2 float64) (float64, error) {
+			if !t1.Before(t2) {
```

Comment:
Is this even a possibility based on how this code works?

Reply:
Correct. It should not be possible. The main thing I wanted to guard against is that somehow a point is sampled multiple times in a window and ends up (maybe artificially by the input) with identical timestamps. That'd give a 0-div error.